queries/dart/textobjects.scm

```diff
@@ -10,20 +14,21 @@
 ; function/method
 (( 
   [(marker_annotation)? (annotation)?] @function.outer.start .
-  [(method_signature) (function_signature)] @_start
-  .
-  (function_body) @_end
+  [(method_signature) (function_signature)] @_start .
+  (function_body) @_end @function.inner
 )
 (make-range! "function.outer" @_start @_end))
-
-(function_body) @function.inner
-(type_alias (function_type) @function.inner) @function.outer
+(type_alias (function_type)? @function.inner) @function.outer
 
 ; parameter
-(formal_parameter) @parameter.inner
-(arguments (_) @parameter.inner)
-(normal_parameter_type) @parameter.inner
-(type_parameter) @parameter.inner
+(formal_parameter_list) @parameter.outer
```

Comment:
parameter.outer is the parameter plus the following comma.

Reply:
Ah, so unless there is a way how to select specific character after parameter (i don't know) this cannot be done with current dart treesitter parser.
Removing outer ones...

Reply:
Ok i found implementation i `c` textobjects. So i added `parameter.outer`.